micro_sam/sam_annotator/_state.py

```diff
@@ -67,3 +68,14 @@ def initialized_for_tracking(self):
                 f"Invalid AnnotatorState: {init_sum} / 2 parts of the state "
                 "needed for tracking are initialized."
             )
+
+    def reset_state(self):
```

Comment:
It seems like it will be easy to accidentally end up with a mixed state situation, where some of the attributes belong to one image, but others are from a different computation.
Eg: We have previously computed some state, including amg and/or tracks state information. Then, we want to compute the 2d image embeddings for a different input image. This will overwrite the `.image_embeddings` and `.predictor` attributes of the state, but because there is no amg or tracking stuff happening now those attributes are left unaltered.
The simplest solution to this problem seems to me that we should totally reset the state every time before we create or load new image embeddings. (Alternatively, we could try to add lines setting each of the unaltered attributes back to None... but that seems easier to make a mistake and accidentally forget some)

Reply:
I think that's a very good idea.